bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/dialogs/ViewContentProvider.java

```diff
@@ -175,18 +174,20 @@ private boolean isIntroView(String id) {
 	}
 
 	/**
-	 * Evaluates if the view is filtered by an activity
+	 * Evaluates if the view is filtered by an activity. Note that this is only
+	 * possible for E3 views, as activities don't exist in the Eclipse 4.
 	 *
-	 * @param elementId
+	 * @param descriptor
 	 * @return result of the check
 	 */
-	private boolean isFilteredByActivity(String elementId) {
-		IViewDescriptor[] views = viewRegistry.getViews();
-		for (IViewDescriptor descriptor : views) {
-			if (descriptor.getId().equals(elementId) && WorkbenchActivityHelper.filterItem(descriptor)) {
-				return true;
-			}
-		}
-		return false;
+	private boolean isFilteredByActivity(MPartDescriptor descriptor) {
+		// viewRegistry.find(...) already applies a filtering for disabled views
+		boolean isFiltered = viewRegistry.find(descriptor.getElementId()) == null;
+
+		// E3 views can be detected by checking whether they use the compatibility layer
+		boolean isE3View = descriptor.getContributionURI().equals(CompatibilityPart.COMPATIBILITY_VIEW_URI);
```

Comment:
In general better to use the constant first, e.g.. CompatibilityPart.COMPATIBILITY_VIEW_URI.equals....

Reply:
Fixed.